src/pip/_internal/utils/outdated.py

```diff
@@ -129,12 +130,10 @@ def pip_version_check(session, options):
                 trusted_hosts=options.trusted_hosts,
                 session=session,
             )
-            all_candidates = finder.find_all_candidates("pip")
-            if not all_candidates:
+            candidate = finder.find_candidates("pip", SpecifierSet()).best
```

Comment:
Oops, somehow my previous reply was lost after rebasing. I asked whether this would be acceptable:

```python
finder.find_candidates("pip", specifier=None)
```

This would make the intention of not filtering explicit without requiring SpecifierSet. (Requiring a specifier string would be fine as well, in this case empty).